src/main/java/edu/harvard/iq/dataverse/harvest/server/OAIRecordServiceBean.java

```diff
@@ -332,17 +339,21 @@ public List<OAIRecord> findOaiRecordsBySetName(String setName, Date from, Date u
         
         if (until != null) { 
             // 24 * 3600 * 1000 = number of milliseconds in a day. 
+                        
+            Date untilDate = Date.from(until);
             
-            if (until.getTime() % (24 * 3600 * 1000) == 0) {
+            if (untilDate.getTime() % (24 * 3600 * 1000) == 0) {
                 // The supplied "until" parameter is a date, with no time
                 // portion. 
+                // TODO: review/reimplement this!
+
                 logger.fine("plain date. incrementing by one day");
-                until.setTime(until.getTime()+(24 * 3600 * 1000));
+                untilDate.setTime(untilDate.getTime()+(24 * 3600 * 1000));
             } else {
                 logger.fine("date and time. incrementing by one second");
-                until.setTime(until.getTime()+1000);
+                untilDate.setTime(untilDate.getTime()+1000);
             }
-            query.setParameter("until",until,TemporalType.TIMESTAMP); 
+            query.setParameter("until",untilDate,TemporalType.TIMESTAMP); 
```

Comment:
**suggestion:** Please not the clock skewing is [done by the XOAI lib already](https://github.com/gdcc/xoai/blob/88b06ef3cb401373b158cb9b71b57286b59db2a3/xoai-data-provider/src/main/java/io/gdcc/xoai/dataprovider/request/RequestBuilder.java#L239).

You can alter the skewing behavior by overriding the `RepositoryConfiguration.skewUntil(()` method if necessary.

Reply:
Sure, we discussed this of course, I forgot to remove it.